0000-catalog.md

```diff
@@ -0,0 +1,801 @@
+are supported:
+
+**catalog actions:**
+
+- CatalogCreate - create a catalog and store it in state
+- CatalogUpdate - update a catalog in state
+- CatalogDelete - deletes a catalog from state and associated catalog_products
+
+**catalog_product actions:**
+
+- CatalogProductCreate - creates a catalog_product and stores it in state
+- CatalogProductUpdate - updates a catalog_product in state
+- CatalogProductDelete - delete a catalog_product from state
+- CatalogProductSetStatus - changes the "status" of a catalog_product to an
+  active, inactive, or discontinued state
+
+The CatalogProductSetStatus action will accept a parameter list of the
+catalogs the action should be performed on. Meaning Activate, Deactivate, or 
+Discontinue can be performed on a single, multiple, or all catalogs that an 
+organization has.
+
+## Permissions
+
+Creation of a Grid Catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field. An organization must have the GS1
+company prefix in the "gs1_company_prefixes" metadata field for its [Pike](https://grid.hyperledger.org/docs/grid/nightly/master/transaction_family_specifications/pike_transaction_family.html) organization.
+
+Deletion of a catalog is restricted to agents acting on behalf of the
+organization in the catalog's owner field. A setting will turn off deletion
+entirely, with the intent that it could be enabled/disabled by a Grid
+administrator. Disabling delete is useful because external systems may have
+references to these products and deleting them could leave dangling references.
+
+_Disclaimer: CatalogDelete & CatalogProductDelete are potentially hazardous
+operation and need to be done with care._
+
+To perform any of the **catalog actions** agents will need the following
+permissions:
+
+```
+CatalogCreate: Agent from org needs the "can_create_catalog" permission
+CatalogUpdate: Agent from org needs the "can_update_catalog" permission
+CatalogDelete: Agent from org needs the "can_delete_catalog" permission
+```
+
+To perform any of the **catalog_product actions** agents inherit the same
+permissions needed for a Grid Product:
+
+```
+CatalogProductCreate: Agent from org needs the "can_create_product" permission
+CatalogProductUpdate: Agent from org needs the "can_update_product" permission
+CatalogProductDelete: Agent from org needs the "can_delete_product" permission
+```
+
+In the case of **catalog operations** agents from the org they represent will
+need the operational permission to perform the desired operation on a Grid
+Catalog.
+
+```
+CatalogProductSetStatus: Agent from org needs the "can_set_catalog_product_status" 
+permission
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## State
+
+### Catalog Representation
+
+The primary object stored in state is `Catalog`, which consists of a
+`catalog_id` (hash of the catalog_name), an `owner` (org_id
```

Comment:
This is actually a hash of the catalog's `name`, right?

Reply:
Correct, the catalog_id is simply a hash of the catalog_name. Collisions in the merkle state are avoided due to the merkle address useing the unique `org_id` as part of its address. It makes it inexpensive to pull all the catalogs associated with a particular org.

Reply:
Oh okay. The `Catalog` message has a field called `name`, but no field called `catalog_name`, which was a bit confusing. I'd recommend updating those field names to match.

Reply:
Updated